components/teams-service/storage/postgres/migration/sql/11_convert_table_to_v2.up.sql

```diff
@@ -0,0 +1,29 @@
+-- under IAM v1, teams were identified by auto-generated UUIDs
+-- under IAM v2, unique IDs are provided by the user
+ALTER TABLE teams DROP COLUMN id;
+
+-- under IAM v1, team names were unique identifiers
+-- under IAM v2, IDs are the unique human-readable identifiers, 
+-- though it is the db_id that is used as the identifier among db tables
+ALTER TABLE teams RENAME COLUMN name TO id;
+
+--- rename uniqueness constraint
+ALTER TABLE teams DROP CONSTRAINT teams_name_key;
+ALTER TABLE teams ADD CONSTRAINT teams_id_key UNIQUE (id);
```

Comment:
💭 there's absolutely no possibility that a customer would have non-unique IDs, right? (Since this would kill upgrading if there was..)

Reply:
we've been storing all IDs under `name` which had a [unique constraint](https://github.com/chef/automate/blob/master/components/teams-service/storage/postgres/migration/sql/01_create-pgstore-table.up.sql#L3) since its creation so I feel pretty confident. 

The main issue customers may run into is that we allowed spaces and non-alphanumeric customers in v1 names. We decided to handle those on a case by case basis rather than try to write a crazy migration